src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -277,10 +349,47 @@ protected override void Initialize()
 
                 _firmata.EnableDigitalReporting();
 
+                foreach (var e in _extendedCommandHandlers)
+                {
+                    e.Registered(_firmata, this);
+                    e.OnConnected();
+                }
+
                 _initialized = true;
             }
         }
 
+        private void RegisterCommandHandlers()
+        {
+            _extendedCommandHandlers.Add(new DhtSensor());
+            _extendedCommandHandlers.Add(new FrequencySensor());
+        }
+
+        /// <summary>
+        /// Registers the known supported modes. Should only be called once from Initialize.
+        /// </summary>
+        private void RegisterKnownSupportedModes()
+        {
+            // We add all known modes to the list, even though we don't really support them all in the core
+            _knownSupportedModes.Add(SupportedMode.DigitalInput);
```

Comment:
Shouldn't all of these happen inside a try after enetring a write lock? I know this happens at initialization, but technically I don't think there is anything preventing folks to try to read supported modes in a different thread while doing initialization in another one.